### DISCUSSION TITLE
DAI 2024-2025 - Practical work 2 - Victor Nicolet, Pierric Ripoll, Colin Moschard

### DISCUSSION THREAD

**Discussion by Nyaaw** [heig-vd-dai-course/heig-vd-dai-course#497] [Show and tell]
Nous voulons faire un petit jeu de calcul mental en console qui utilise TCP et UDP. Le déroulement du jeu se passe comme ça:

- Serveur envoie un broadcast UDP : "Attente de connection", il attend la connection de nouveaux joueurs pendant 1 minute.
- Connection des clients par TCP au serveur, le serveur accepte leur connections.
- À la fin de la minute d'attente, le serveur lance le jeu. Le jeu est d'une durée limitée
- Pour chaque client, le jeu commence : 
  - le serveur envoie un calcul mental (ex. "1242 + 18 *  5 - 240")
  - le client envoie une réponse
  - si c'est juste, le serveur incrémente les points du client et envoie le prochain calcul, sinon, le serveur envoie un message "faux" et attend la bonne réponse.
- à la fin du temps imparti, le serveur ferme les connections TCP, envoie les résultats en broadcast UDP, et attend les connections des prochains joueurs.

**Comment by ludelafo** [comment]
Tout bon pour nous !

**Comment by colinmoschard** [comment]
Repository: https://github.com/[Vicolet/OPrun](https://github.com/Vicolet/OPrun)
Commit: 0b1620c
